### PR TITLE
[security/tinc] Enable usage of no, one or several addresses per hosts

### PR DESCRIPTION
**Pull request by ElNounch**
All three situations are legit, both for local and remote hosts.

### CODE REVIEW

File: security/tinc/src/opnsense/scripts/OPNsense/Tinc/lib/objects.py [hunk 1]
```diff
@@ -97,6 +95,10 @@ def set_PMTUDiscovery(self, value):
     def config_text(self):
         result = list()
         result.append('AddressFamily=any')
+        if 'address' in self._payload:
+            addresses = self._payload['address'].split(',')
+            for address in addresses:
```
Comment (AdSchellevis) [inline security/tinc/src/opnsense/scripts/OPNsense/Tinc/lib/objects.py:100]: can be simplified:

```
for address in self._payload['address'].split(','):
```

`addresses`  isn't used anywhere else.

It's probably better to move the functionality to `NetwConfObject. get_addresses()` in which case you could iterate always over the get_addresses() and ditch the `if` as well

File: security/tinc/src/opnsense/scripts/OPNsense/Tinc/lib/objects.py [hunk 2]
```diff
@@ -137,7 +139,10 @@ def set_connectto(self, value):
 
     def config_text(self):
         result = list()
-        result.append('Address=%(address)s %(port)s'%self._payload)
+        if 'address' in self._payload:
+            addresses = self._payload['address'].split(',')
+            for address in addresses:
```
Comment (AdSchellevis) [inline security/tinc/src/opnsense/scripts/OPNsense/Tinc/lib/objects.py:144]: same

File: security/tinc/src/opnsense/mvc/app/models/OPNsense/Tinc/Tinc.xml [hunk 3]
```diff
@@ -22,7 +22,7 @@
                     <ValidationMessage>Please specify a valid hostname.</ValidationMessage>
                 </hostname>
                 <extaddress type="TextField">
-                    <Required>Y</Required>
```
Comment (AdSchellevis) [inline security/tinc/src/opnsense/mvc/app/models/OPNsense/Tinc/Tinc.xml:25]: I think we could also use a `HostnameField` in combination with a tokeniser in the form, in which case you wouldn't need to explain the split character.

**Review by AdSchellevis — CHANGES_REQUESTED** [review]

File: security/tinc/src/opnsense/scripts/OPNsense/Tinc/lib/objects.py (hunk 1 above)
Comment (ElNounch) [reply security/tinc/src/opnsense/scripts/OPNsense/Tinc/lib/objects.py]: Was micmic of existing code for subnets...
Reworked both this way.

File: security/tinc/src/opnsense/scripts/OPNsense/Tinc/lib/objects.py (hunk 2 above)
Comment (ElNounch) [reply security/tinc/src/opnsense/scripts/OPNsense/Tinc/lib/objects.py]: Was micmic of existing code for subnets...
Reworked both this way.

File: security/tinc/src/opnsense/scripts/OPNsense/Tinc/lib/objects.py (hunk 1 above)
Comment (ElNounch) [reply security/tinc/src/opnsense/scripts/OPNsense/Tinc/lib/objects.py]: Did so, but this part shouldn't have been, anyway...
Undone. Took care of your comment in next code block.

File: security/tinc/src/opnsense/scripts/OPNsense/Tinc/lib/objects.py (hunk 2 above)
Comment (AdSchellevis) [reply security/tinc/src/opnsense/scripts/OPNsense/Tinc/lib/objects.py]: @ElNounch nice!

**Comment by AdSchellevis** [comment]
@ElNounch did a small test on my end, looks good, I only added a new version number in https://github.com/opnsense/plugins/commit/5dc2478cd7e31069b6a5fadb8750a1fc6cfb20b8 and merged the PR.

**Pull request merged**